bundles/org.openhab.voice.porcupineks/src/main/java/org/openhab/voice/porcupineks/internal/PorcupineKSService.java

```diff
@@ -0,0 +1,318 @@
+    @Override
+    public Set<Locale> getSupportedLocales() {
+        return Set.of(Locale.ENGLISH, new Locale("es"), Locale.FRENCH, Locale.GERMAN);
+    }
+
+    @Override
+    public Set<AudioFormat> getSupportedFormats() {
+        return Set
+                .of(new AudioFormat(AudioFormat.CONTAINER_WAVE, AudioFormat.CODEC_PCM_SIGNED, false, 16, null, 16000L));
+    }
+
+    @Override
+    public KSServiceHandle spot(KSListener ksListener, AudioStream audioStream, Locale locale, String keyword)
+            throws KSException {
+        Porcupine porcupine;
+        if (config.apiKey.isBlank()) {
+            throw new KSException("Missing pico voice api key");
+        }
+        BundleContext bundleContext = this.bundleContext;
+        if (bundleContext == null) {
+            throw new KSException("Missing bundle context");
+        }
+        try {
+            porcupine = initPorcupine(bundleContext, locale, keyword);
+        } catch (PorcupineException | IOException e) {
+            throw new KSException(e);
+        }
+        Future<?> scheduledTask = executor.submit(() -> processInBackground(porcupine, ksListener, audioStream));
+        return () -> {
+            loop = false;
+            scheduledTask.cancel(true);
+            porcupine.delete();
+        };
+    }
+
+    private Porcupine initPorcupine(BundleContext bundleContext, Locale locale, String keyword)
+            throws IOException, PorcupineException {
+        // Suppress library logs
+        java.util.logging.Logger globalJavaLogger = java.util.logging.Logger
+                .getLogger(java.util.logging.Logger.GLOBAL_LOGGER_NAME);
+        Level currentGlobalLogLevel = globalJavaLogger.getLevel();
+        globalJavaLogger.setLevel(java.util.logging.Level.OFF);
+        String bundleLibraryPath = Porcupine.LIBRARY_PATH;
+        if (bundleLibraryPath == null) {
+            throw new PorcupineException("Unsupported environment, ensure Porcupine is supported by your system");
+        }
+        String libraryPath = prepareLib(bundleContext, bundleLibraryPath);
+        String alternativeModelPath = getAlternativeModelPath(bundleContext, locale);
+        String modelPath = alternativeModelPath != null ? alternativeModelPath
+                : prepareLib(bundleContext, Porcupine.MODEL_PATH);
+        String keywordPath = getKeywordResourcePath(bundleContext, keyword, alternativeModelPath == null);
+        logger.debug("Porcupine library path: {}", libraryPath);
+        logger.debug("Porcupine model path: {}", modelPath);
+        logger.debug("Porcupine keyword path: {}", keywordPath);
+        logger.debug("Porcupine sensitivity: {}", config.sensitivity);
+        try {
+            return new Porcupine(config.apiKey, libraryPath, modelPath, new String[] { keywordPath },
+                    new float[] { config.sensitivity });
+        } finally {
+            // restore log level
+            globalJavaLogger.setLevel(currentGlobalLogLevel);
+        }
```

Comment:
Your idea is to restore the log level when the object will be destroyed ?
I have a big doubt here (I am not very familiar with this use case), maybe the code will be run just after the return ?

Reply:
No, the finally clause get executed before the function returns the value, just after the call to the Porcupine constructor.
So the log is only disabled for the execution of this method.

Reply:
Ok, I did not understand this was what er are trying to achieve.

Reply:
Why not in this case simply do it like that (which looks clearer):
```
Porcupine p = new Porcupine(config.apiKey, libraryPath, modelPath, new String[] { keywordPath },
                    new float[] { config.sensitivity });
// restore log level
globalJavaLogger.setLevel(currentGlobalLogLevel);
return p;
```

Reply:
@lolodomo - if `Porcupine` constructor would throw an unchecked exception for any reason, log level would not be restored. Using `finally` is more robust as it will always be executed, even if an exception is thrown.

Reply:
Ok, you're right.